cmake/variables.cmake

```diff
@@ -166,7 +166,7 @@ if(NOT "$ENV{ML_DEBUG}")
       list(APPEND ML_COMPILE_DEFINITIONS "_FORTIFY_SOURCE=2" "_REENTRANT")
```

Comment:
I just noticed something else that's wrong here: `_REENTRANT` should always be defined when building for Linux, whether it's a debug build or not.

In other words, `_REENTRANT` should be added to the compile definitions in a block that's only guarded by `CMAKE_SYSTEM_NAME STREQUAL "Linux"`.

Since this is a Windows PR I'm happy to leave this for a followup, but it needs fixing at some point.